store/blockstore_test.go

```diff
@@ -0,0 +1,45 @@
+package store
+
+import (
+	"testing"
+
+	"github.com/lazyledger/optimint/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlockstoreHeight(t *testing.T) {
+	assert := assert.New(t)
+
+	bstore := NewBlockStore()
+	assert.Equal(uint64(0), bstore.Height())
+
+	err := bstore.SaveBlock(&types.Block{
+		Header: types.Header{
+			Height: 1,
+		},
+	})
+	assert.NoError(err)
+	assert.Equal(uint64(1), bstore.Height())
```

Comment:
Better to use table-driven tests and test more cases (instead of only height == 1). Especially edge-cases are interesting too.

Reply:
I think it makes sense to address this in this PR too.

Reply:
Improved tests are almost ready.